syntax = "proto3";

package proto.gradido;

import "gradido/BasicTypes.proto";

/*
	id will be set by Node server
	txHash will be also set by Node server,
	  calculated from previous transaction txHash and this id, sigMap and received;
*/



message GradidoTransaction {
	SignatureMap sig_map = 1;

    // TODO: consider if having concrete type wouldn't be better; 
    // having bytes would better show what is signed, still for 
    // blockchain it doesn't matter, as it has to be serialized again
    // when validating
	// UNICORN ANSWER:
	// Maybe but hedera has at first concrete type and has depracted that now, so is must 
	// be reason for that
	bytes body_bytes = 2; // TransactionBody
}
